scripts/stats-aggregator/daily.ts

```diff
@@ -0,0 +1,143 @@
+import commandLineArgs from "command-line-args";
+import commandLineUsage from "command-line-usage";
+
+import dayjs from "@blobscan/dayjs";
+import { normalizeDailyDatePeriod, prisma, RawDatePeriod } from "@blobscan/db";
+
+import { ALL_ENTITIES, deleteOptionDef, Entity, helpOptionDef } from "./common";
+
+const dailyCommandOptDefs: commandLineUsage.OptionDefinition[] = [
+  deleteOptionDef,
+  helpOptionDef,
+  {
+    name: "entity",
+    alias: "e",
+    typeLabel: "{underline type}",
+    description:
+      "Entity type to aggregate. Valid values are {italic blob}, {italic block} or {italic tx}.",
+    type: String,
+    multiple: true,
+  },
+  {
+    name: "from",
+    alias: "f",
+    typeLabel: "{underline date}",
+    description: "Start date in ISO 8601 format.",
+    type: String,
+  },
+  {
+    name: "to",
+    alias: "t",
+    typeLabel: "{underline date}",
+    description: "End date in ISO 8601 format.",
+    type: String,
+  },
+];
+
+const dailyCommandUsage = commandLineUsage([
+  {
+    header: "Daily Command",
+    content: "Aggregate daily stats.",
+  },
+  {
+    header: "Options",
+    optionList: dailyCommandOptDefs,
+  },
+]);
+
+export async function performDailyStatsOperation(
+  entity: Entity,
+  operation: "populate" | "deleteMany",
+  rawDatePeriod?: RawDatePeriod
+) {
+  let dailyStatsFn;
+  const datePeriodProvided = rawDatePeriod?.from || rawDatePeriod?.to;
+  const { from, to } = normalizeDailyDatePeriod(rawDatePeriod);
+  const operation_ =
+    operation === "deleteMany" && !datePeriodProvided ? "deleteAll" : operation;
+  const operationParam =
+    operation_ === "populate"
+      ? rawDatePeriod
+      : datePeriodProvided
+      ? {
+          where: {
+            day: {
+              ...(from ? { gte: from } : {}),
+              ...(to ? { lte: to } : {}),
+            },
+          },
+        }
+      : undefined;
+
+  switch (entity) {
+    case "blob":
+      dailyStatsFn = prisma.blobDailyStats[operation_];
+      break;
+    case "block":
+      dailyStatsFn = prisma.blockDailyStats[operation_];
+      break;
+    case "tx":
+      dailyStatsFn = prisma.transactionDailyStats[operation_];
+      break;
+  }
+
+  const result = await dailyStatsFn(operationParam).then((res) => {
+    if (operation_ === "deleteAll") {
+      return "All";
+    }
+
+    return typeof res === "number" ? res : res.count;
+  });
+
+  const formattedFrom = from
+    ? dayjs(from).format("DD-MM-YYYY")
```

Comment:
I'd go for using YYYY-MM-DD as it is universal and can be compared using "<" and ">" operators

Reply:
Yes. It should follow that format. I'll fix it